### PR TITLE
Iceberg cleanup

### PR DESCRIPTION
**Pull request by electrum**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

refactoring

## Related issues, pull requests, and links

These are preparatory commits for https://github.com/trinodb/trino/pull/13219 and https://github.com/trinodb/trino/pull/7933

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java [hunk 1]
```diff
@@ -236,7 +236,7 @@ public void deleteRows(Block rowIds)
             positionDeleteSink = positionDeleteSinkSupplier.get();
             verify(positionDeleteSink != null);
         }
-        positionDeleteSink.appendPage(new Page(rowIds));
+        positionDeleteSink.appendPage(new Page(rowIds)).join();
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java:239]: > Fix Error Prone warnings in Iceberg

looks like some warning that we would want to have on error level perhaps

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java [hunk 2]
```diff
@@ -556,12 +555,15 @@ else if (orcColumn != null) {
...
-        catch (Exception e) {
+        catch (IOException | RuntimeException e) {
             if (orcDataSource != null) {
                 try {
                     orcDataSource.close();
                 }
-                catch (IOException ignored) {
+                catch (IOException ex) {
+                    if (e.equals(ex)) {
+                        e.addSuppressed(ex);
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java:565]: ! equals

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java [hunk 3]
```diff
@@ -898,7 +900,10 @@ else if (column.isRowPositionColumn()) {
                     dataSource.close();
                 }
             }
-            catch (IOException ignored) {
+            catch (IOException ex) {
+                if (e.equals(ex)) {
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java:904]: ! equals

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTransforms.java [hunk 4]
```diff
@@ -710,7 +710,7 @@ private static Block transformBlock(Type sourceType, FixedWidthType resultType,
     @VisibleForTesting
     static long epochYear(long epochMilli)
     {
-        return YEAR_FIELD.get(epochMilli) - 1970;
+        return YEAR_FIELD.get(epochMilli) - 1970L;
```
Comment (findepi) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTransforms.java:713]: cheat

**Review by findepi — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java (hunk 2 above)
Comment (electrum) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java:565]: Good catch

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java (hunk 1 above)
Comment (electrum) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSource.java:239]: Filed https://github.com/trinodb/trino/issues/13266

**Pull request merged**